### PR TITLE
St/enhancement-syringe_codes

### PR DESCRIPTION
**Pull request by sean-v8**
## M382 + M383 - Extend/retract syringe on specified tool

eg. `M382 T1` to drop silver on T1, `M383 T1` to retract it

### Requirements
- [x] Alignment run successfully
- [x] Free Ram looks reasonable
- [x] Approval 1
- [x] Approval 2

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -5450,6 +5450,83 @@ inline void gcode_M303() {
...
+    if (code_seen('T')) {
+      tool = code_value();
+    }
+    switch(tool) {
+        case 0:
+          OUT_WRITE(SYRINGE0_PIN, LOW);
+          break;
+        case 1:
+          OUT_WRITE(SYRINGE1_PIN, LOW);
+          break;
```
Comment (pizzyflavin) [inline Marlin/Marlin_main.cpp]: Note to self: Check that `OUT_WRITE()` can handle a macro as an argument

Comment (sean-v8) [inline Marlin/Marlin_main.cpp]: It can.

Comment (pizzyflavin) [inline Marlin/Marlin_main.cpp]: I remember now what it was. It can't take a variable as an argument I believe. 👟

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -5462,9 +5462,13 @@ inline void gcode_M303() {
...
-          OUT_WRITE(SYRINGE0_PIN, HIGH);
-          current_syringe_pin = SYRINGE0_PIN;
-          break;
+          if (!Cartridge__GetAugerEnabled()) {
+            OUT_WRITE(SYRINGE0_PIN, HIGH);
+            current_syringe_pin = SYRINGE0_PIN;
+            break;
+          }
+          SERIAL_PROTOCOLPGM("Auger enabled, syringe extension canceled");
+          return;
```
Comment (pizzyflavin) [inline Marlin/Marlin_main.cpp]: Break should be outside the if, at the end of the case. I'd put this warning message in the `if` statement above and put the expected behavior here (i.e. switch them and remove the `!`). Same for `M383` below.

File: Marlin/Marlin_main.cpp [hunk 3]
```diff
@@ -5450,6 +5450,91 @@ inline void gcode_M303() {
...
+        case 1:
+          OUT_WRITE(SYRINGE1_PIN, HIGH);
+          current_syringe_pin = SYRINGE1_PIN;
+          break;
+      // Invalid Tool Number
+      default:
+        SERIAL_ECHO_START;
+        SERIAL_CHAR('T');
```
Comment (pizzyflavin) [inline Marlin/Marlin_main.cpp]: We should probably change `SERIAL_CHAR()` TO `SERIAL_PROTOCOLPGM()`

File: Marlin/Marlin_main.cpp [hunk 4]
```diff
@@ -5450,6 +5450,91 @@ inline void gcode_M303() {
...
+          }
+          SERIAL_PROTOCOLPGM("Auger enabled, syringe retraction canceled");
+          return;
+        case 1:
+          OUT_WRITE(SYRINGE1_PIN, LOW);
+          break;
+      // Invalid Tool Number
+      default:
+        SERIAL_ECHO_START;
+        SERIAL_CHAR('T');
```
Comment (pizzyflavin) [inline Marlin/Marlin_main.cpp]: Change to `SERIAL_PROTOCOLPGM()`

**Comment by danthompson41** [comment]
💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯

**Pull request merged**